File: backend/internal/dto/request/address.go
package request

type AddCryptoAddressReq struct {
	Currency string `json:"currency" binding:"required"`
	Chain    string `json:"chain" binding:"required"`
	Address  string `json:"address" binding:"required"`
	Alias    string `json:"alias" binding:"required"`
}

type AddBankAccountReq struct {
	Currency           string `json:"currency" binding:"required"`
	BankName           string `json:"bank_name" binding:"required"`
	BankCountry        string `json:"bank_country" binding:"required"`
	SwiftCode          string `json:"swift_code"`
	AccountName        string `json:"account_name" binding:"required"`
	AccountNo          string `json:"account_no" binding:"required"`
	TransferType       string `json:"transfer_type" binding:"required"`
	AccountType        string `json:"account_type"`
	PayeeCountryCode   string `json:"payee_country_code"`
	PayeeAddress       string `json:"payee_address"`
	PayeeAddressSecond string `json:"payee_address_second"`
	BankCode           string `json:"bank_code"`
	BankAddress        string `json:"bank_address"`
	MiddleSwiftCode    string `json:"middle_swift_code"`
}

File: frontend/src/types/address.ts
export interface BankAccount {
  id: number;
  currency: string;
  bank_name: string;
  bank_country: string;
  swift_code: string;
  account_name: string;
  account_no_masked: string;
  transfer_type: string;
  status: string;
}

export interface AddBankAccountRequest {
  currency: string;
  bank_name: string;
  bank_country: string;
  swift_code?: string;
  account_name: string;
  account_no: string;
  transfer_type: string;
  account_type?: string;
  payee_country_code?: string;
  payee_address?: string;
  payee_address_second?: string;
  bank_code?: string;
  bank_address?: string;
  middle_swift_code?: string;
}
